src/Core/Error.idr

```diff
@@ -0,0 +1,53 @@
+module Core.Error
+
+import Core.Core
+import Core.Context
+
+export
+tryError : {auto c : Ref Ctxt Defs} ->
+           {auto refs : RefList ls} ->
```

Comment:
I'm not sure how this `auto` is supposed to be inferred.
Should it (or `ls`?) be an explicit argument instead?